x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/selectors.ts

```diff
@@ -4,6 +4,68 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import qs from 'querystring';
 import { AlertListState } from '../../types';
 
+/**
+ * Returns the Alert Data array from state
+ */
 export const alertListData = (state: AlertListState) => state.alerts;
+
+/**
+ * Returns the alert list pagination data from state
+ */
+export const alertListPagination = (state: AlertListState) => {
+  return {
+    pageIndex: state.request_page_index,
+    pageSize: state.request_page_size,
+    resultFromIndex: state.result_from_index,
+    total: state.total,
+  };
+};
+
+/**
+ * Returns a boolean based on whether or not the user is on the alerts page
+ */
+export const isOnAlertPage = (state: AlertListState): boolean => {
+  return state.location ? state.location.pathname.endsWith('/alerts') : false;
+};
+
+/**
+ * Returns the query object received from parsing the URL query params
+ */
+export const paginationDataFromUrl = (state: AlertListState): qs.ParsedUrlQuery => {
+  // Removes the `?` from the beginning of query string if it exists
+  return state.location ? qs.parse(state.location.search.slice(1)) : {};
```

Comment:
i think you need to call out / validate each value here. otherwise a malicious actor can give an unwitting nerd a url like `somekibana.net/endpoint?badStuff=encodedHacks` and then the nerd clicks the url and a request goes to kibana w/ unwitting nerd's creds but with whatever params were provided by malicious actor @dplumlee #metaprogramming-is-antipattern

Reply:
@oatkiller We are doing pretty comprehensive validation on the backend FYI. So maybe not necessary? Always helps to have an extra layer, I s'pose.

Reply:
I was also wondering this. I think once we start to grab data from the URL which is open to manipulation, we should do some sort of manipulation in order to avoid front-end code injections. I'm ok if we defer that to another Issue, but we should come up with a common approach.